homeassistant/components/home_connect/__init__.py

```diff
@@ -615,14 +615,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: HomeConnectConfigEntry)
     home_connect_client = HomeConnectClient(config_entry_auth)
 
     coordinator = HomeConnectCoordinator(hass, entry, home_connect_client)
-    await coordinator.async_config_entry_first_refresh()
-
     entry.runtime_data = coordinator
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     entry.runtime_data.start_event_listener()
 
+    entry.async_create_task(
+        hass,
+        coordinator.async_config_entry_first_refresh(),
+        f"home_connect-initial-full-refresh-{entry.entry_id}",
+    )
```

Comment:
We can't use `async_config_entry_first_refresh` since the task won't be executed inside `async_setup_entry`.
```suggestion
    entry.async_create_background_task(
        hass,
        coordinator.async_refresh(),
        f"home_connect-initial-full-refresh-{entry.entry_id}",
    )
```